### PR TITLE
feat: config analysis description improvements

### PR DESCRIPTION
**Pull request by Prashanth-Nelli**

### CODE REVIEW

File: src/components/ConfigInsights/index.tsx [hunk 1]
```diff
@@ -39,6 +49,13 @@ function ConfigInsightsDetails({ configID }: Props) {
         `/api/db/config_analysis?config_id=eq.${configID}`
       );
       const data = (await res.json()) as ConfigTypeInsights[];
+      data.forEach((item) => {
```
Comment (mainawycliffe) [inline src/components/ConfigInsights/index.tsx:52]: Just nitpicking here; we should avoid mutations like this, as they can introduce bugs that are very unpredictable and very hard to track (Javascript passes objects by reference, so you may be mutating something being used somewhere else). Instead, use the .map array function, which is the same number of lines but produces a new object.

File: src/utils/common.js [hunk 2]
```diff
@@ -32,3 +33,26 @@ export function toQueryString(paramsObject) {
     )
     .join("&");
 }
+
+export const sanitizeHTMLContent = (htmlString) => {
```
Comment (mainawycliffe) [inline src/utils/common.js:37]: Can we please add types for these input parameters? We will turn on typescript strictness in the future, and we need to make this a habit.

File: src/utils/common.js [hunk 3]
```diff
@@ -32,3 +33,26 @@ export function toQueryString(paramsObject) {
...
+  return sanitizeHtml(htmlString, {
+    allowedTags: sanitizeHtml.defaults.allowedTags.concat(["img"]),
+    allowedAttributes: {
+      "*": ["style", "class"]
+    }
+  });
+};
+
+export const sanitizeHTMLContentToText = (htmlString) => {
```
Comment (mainawycliffe) [inline src/utils/common.js:46]: Same for this as well

File: src/utils/common.js [hunk 4]
```diff
@@ -32,3 +33,26 @@ export function toQueryString(paramsObject) {
...
+  const html = sanitizeHTMLContent(htmlString);
+  const div = document.createElement("div");
+  div.innerHTML = html;
+  return div.textContent;
+};
+
+export const truncateText = (txt = "", limit) => {
```
Comment (mainawycliffe) [inline src/utils/common.js:53]: And this as well.

**Review by mainawycliffe — COMMENTED** [review]
Just a few points to note otherwise looks good.

**Review by mainawycliffe — APPROVED** [review]

**Comment by moshloop** [comment]
@Prashanth-Nelli  - we are already using dompurify ?

File: src/utils/common.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { parse } from "qs";
+import sanitizeHtml from "sanitize-html";
+
+export function toFixedIfNecessary(value: string, dp: number) {
+  return +parseFloat(value).toFixed(dp);
+}
+
+export function capitalizeFirstLetter(string: string) {
```
Comment (moshloop) [inline src/utils/common.ts:8]: Isn't this a css transform ?

File: src/utils/common.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  return `${parseFloat((bytes / k ** i).toFixed(dm))} ${sizes[i]}`;
+}
+
+export function searchParamsToObj(searchParams: Record<string, string>) {
+  return parse(searchParams.toString());
+}
+
+export function toQueryString(paramsObject: Record<string, string>) {
+  return Object.keys(paramsObject)
```
Comment (moshloop) [inline src/utils/common.ts:29]: Should we use `qs` for this ?

File: src/utils/common.ts (hunk 5 above)
Comment (Prashanth-Nelli) [reply src/utils/common.ts:8]: we have a use case where we capitalise a specific words in a sentence. css transform doesn't work in that case.

**Comment by Prashanth-Nelli** [comment]
> @Prashanth-Nelli - we are already using dompurify ?

I'll switch to this module and remove sanitize-html dependency.

**Pull request merged**